cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -81,23 +81,11 @@ func startNodeIpamController(ccmConfig *cloudcontrollerconfig.CompletedConfig, n
 	// the CloudAllocator ipam does not need service or cluster cidrs since it obtains the data from the cloud directly
 	// for backward compatiblity reasons we don't fail if there are any of those values configured
 	if ipam.CIDRAllocatorType(ccmConfig.ComponentConfig.KubeCloudShared.CIDRAllocatorType) != ipam.CloudAllocatorType {
-		// failure: bad cidrs in config
-		cidrs, dualStack, err := processCIDRs(ccmConfig.ComponentConfig.KubeCloudShared.ClusterCIDR)
+		// process CIDR from config and validate with stack type
+		clusterCIDRs, err := validClusterCIDR(ccmConfig.ComponentConfig.KubeCloudShared.ClusterCIDR)
```

Comment:
it seems the problem is the ccm is crashlooping

> Dec 20 19:34:04.806521 kt2-250fada7-2fbe-master kubelet[2419]: I1220 19:34:04.806289    2419 status_manager.go:693] "Ignoring same status for pod" pod="kube-system/cloud-controller-manager-kt2-250fada7-2fbe-master" status={"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2023-12-20T19:33:20Z"},{"type":"Ready","status":"False","lastProbeTime":null,"lastTransitionTime":"2023-12-20T19:34:00Z","reason":"ContainersNotReady","message":"containers with unready status: [cloud-controller-manager]"},{"type":"ContainersReady","status":"False","lastProbeTime":null,"lastTransitionTime":"2023-12-20T19:34:00Z","reason":"ContainersNotReady","message":"containers with unready status: [cloud-controller-manager]"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2023-12-20T19:33:20Z"}],"hostIP":"10.128.0.2","podIP":"10.128.0.2","podIPs":[{"ip":"10.128.0.2"}],"startTime":"2023-12-20T19:33:20Z","containerStatuses":[{"name":"cloud-controller-manager","state":{"waiting":{"reason":"CrashLoopBackOff","message":"back-off 20s restarting failed container=cloud-controller-manager pod=cloud-controller-manager-kt2-250fada7-2fbe-master_kube-system(6e92b521ff2c7e623ea26a5f986be6a8)"}},"lastState":{"terminated":{"exitCode":1,"reason":"Error","startedAt":"2023-12-20T19:33:57Z","finishedAt":"2023-12-20T19:33:59Z","containerID":"containerd://5ce601b2a647a431d94702710c80075ad03175b2a0e0c9e50a9463175192c3d4"}},"ready":false,"restartCount":2,"image":"registry.k8s.io/cloud-controller-manager:v1.28.2-alpha_5c2c24254bec7c9c96d58f309623a1d00815df8a","imageID":"sha256:ccd18be1e19e82d671dd87d25f5a6060a262b2ecdcda87db8f4fce5ff393125a","containerID":"containerd://5ce601b2a647a431d94702710c80075ad03175b2a0e0c9e50a9463175192c3d4","started":false}],"qosClass":"Burstable"}

it means something was not failing before at startup is failing now, assuming it keeps runing this controller, this seems to be the only place that changes , is this change keeping the validation the same or is stricter now?, can we try keeping this code as it was before just to test?

Reply:
just don't consolidate code, duplicate it in the other controller if necessary, but see if it fails if we only modify the gknecontroller

Reply:
Thanks for the fast investigation @aojea ! I duplicated the validation for `cluster-cidr` with the new commit.

Reply:
It passes all the tests now! Great thanks @aojea! 

hi @sdmodi I will put the re-create default GNP logic back and ping you for review later.

Reply:
I will keep both controllers as much independent as possible, since this is an OSS component we don't want to create dependencies that may cause much complicated the maintenance later